prow/cmd/deck/main.go

```diff
@@ -398,7 +399,7 @@ func localOnlyMain(cfg config.Getter, o options, mux *http.ServeMux) *http.Serve
 	mux.Handle("/github-login", gziphandler.GzipHandler(handleSimpleTemplate(o, cfg, "github-login.html", nil)))
 
 	if o.spyglass {
-		initSpyglass(cfg, o, mux, nil)
+		initSpyglass(cfg, o, mux, nil, nil, nil)
```

Comment:
nit: methods that get called with three  `nil` arguments smell

Reply:
Well yeah, I did not introduce the idea of having a spyglass version without secrets for development, I just adhered to what I found.

Reply:
/assign @Katharine
your thoughts?

Reply:
I think it's probably fine. We absolutely want it to work without any secrets or other authentication (and we want it to be fully functional in this state).

`nil, nil, nil` isn't the best but 🤷‍♀